### PR TITLE
433 : Improve performance of plans list endpoint

### PR DESCRIPTION
**Pull request by rehammuzzamil**
https://github.com/OpenSRP/opensrp-server-web/issues/433

### CODE REVIEW

File: src/test/java/org/opensrp/repository/postgres/PlanRepositoryTest.java [hunk 1]
```diff
@@ -630,4 +625,66 @@ public void testCountAllPlanShouldReturnCorrectValue(){
...
+
+        PlanSearchBean planSearchBean = new PlanSearchBean();
+        Map<String, String> useContextSearchBean = new HashMap<>();
+        useContextSearchBean.put("interventionType", "FI");
+        planSearchBean.setUseContexts(useContextSearchBean);
+        planSearchBean.setExperimental(true);
+
+        List<PlanDefinition> planDefinitions = planRepository.getAllPlans(planSearchBean);
+        assertEquals(1l, planDefinitions.size());
+        assertEquals(1l, planDefinitions.get(0).getUseContext().size());
```
Comment (githengi) [inline src/test/java/org/opensrp/repository/postgres/PlanRepositoryTest.java:687]: add asserts for plan identifier

File: src/test/java/org/opensrp/repository/postgres/PlanRepositoryTest.java [hunk 2]
```diff
@@ -630,4 +625,66 @@ public void testCountAllPlanShouldReturnCorrectValue(){
...
+        plan.setJurisdiction(jurisdictions);
+        plan.setServerVersion(1000l);
+        plan.setExperimental(true);
+        planRepository.add(plan);
+
+        PlanSearchBean planSearchBean = new PlanSearchBean();
+        planSearchBean.setExperimental(true);
+
+        List<PlanDefinition> planDefinitions = planRepository.getAllPlans(planSearchBean);
+        assertEquals(1l, planDefinitions.size());
```
Comment (githengi) [inline src/test/java/org/opensrp/repository/postgres/PlanRepositoryTest.java:654]: add asserts for plan identifier

File: src/test/java/org/opensrp/service/PlanServiceTest.java [hunk 3]
```diff
@@ -63,8 +64,9 @@ public void setUp() {
 	
 	@Test
 	public void testGetAllPlansShouldCallRepositoryGetAllMethod() {
-		planService.getAllPlans(false);
-		verify(planRepository).getAllPlans(eq(false));
+		PlanSearchBean planSearchBean = new PlanSearchBean();
+		planService.getAllPlans(planSearchBean);
+		verify(planRepository).getAllPlans(any(PlanSearchBean.class));
```
Comment (githengi) [inline src/test/java/org/opensrp/service/PlanServiceTest.java:69]:
```suggestion
		verify(planRepository).getAllPlans(planSearchBean);
```

File: src/test/java/org/opensrp/repository/postgres/PlanRepositoryTest.java (hunk 2 above)
Comment (rehammuzzamil) [reply src/test/java/org/opensrp/repository/postgres/PlanRepositoryTest.java:654]: Added

File: src/test/java/org/opensrp/repository/postgres/PlanRepositoryTest.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/test/java/org/opensrp/repository/postgres/PlanRepositoryTest.java:688]: Added

**Review by githengi — APPROVED** [review]

**Pull request merged**